driver-mapping/src/main/java/com/datastax/driver/mapping/config/PropertyAccessStrategy.java

```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * Returns whether or not the given {@link Field} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given field should be considered, {@code false} otherwise.
+     */
+    boolean isFieldAccessAllowed();
+
+    /**
+     * Returns whether or not the given {@link PropertyDescriptor} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given {@link PropertyDescriptor} should be considered, {@code false} otherwise.
+     */
+    boolean isGetterSetterAccessAllowed();
+
+    /**
+     * Locates a getter method for the given base class and given property.
+     * If this method returns {@code null} then no getter will be used to access the given property.
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                  either {@link com.datastax.driver.mapping.annotations.Table @Table} or
+     *                  {@link com.datastax.driver.mapping.annotations.UDT @UDT}.
+     * @param property  The property to locate a getter for; never {@code null}.
+     * @return The getter method for the given base class and given property, or {@code null} if no getter was found.
+     */
+    Method locateGetter(Class<?> mappedClass, PropertyDescriptor property);
+
+    /**
+     * Locates a setter method for the given base class and given property.
+     * If this method returns {@code null} then no setter will be used to access the given property.
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                  either {@link com.datastax.driver.mapping.annotations.Table @Table} or
+     *                  {@link com.datastax.driver.mapping.annotations.UDT @UDT}.
+     * @param property  The property to locate a setter for; never {@code null}.
+     * @return The setter method for the given base class and given property, or {@code null} if no setter was found.
+     */
+    Method locateSetter(Class<?> mappedClass, PropertyDescriptor property);
+
+    /**
+     * Reads a property.
+     *
+     * @param entity       The instance to read the property from; never {@code null}.
+     * @param propertyName The property name, as inferred by the mapper; never {@code null}.
+     * @param field        The property field; may be {@code null}.
+     *                     May be {@code null}, but {@code field} and {@code getter} cannot be both {@code null}.
+     * @param getter       The property's getter method, as inferred by {@link #locateGetter(Class, PropertyDescriptor) this strategy}.
+     *                     May be {@code null}, but {@code field} and {@code getter} cannot be both {@code null}.
+     * @return The property value.
+     * @throws IllegalArgumentException if the property cannot be read.
+     */
+    Object getValue(Object entity, String propertyName, Field field, Method getter);
```

Comment:
Sorry, I edited my previous comment to mention that I was leaning towards the idea ofkeeping `getValue` and `setValue`. 

I am still working on it, but cf. my previous comment, this might be too far-fetched. To let users really bypass reflection altogether – which is a plausible goal after all – we need a lot more refactoring. That's why I think it goes beyond the scope of this PR. But if I end up with something that works and passes the tests during the weekend, I might open a separate PR just to let you reviewers compare and judge  by yourselves.

In the meanwhile feel free to revert my commits or add yours on top of them as you deem useful.